lib/valkyrie/specs/shared_specs/supports_ordered_persisting.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+RSpec.shared_examples 'it supports persisting ordered properties' do
+  before do
+    raise 'persister must be set with `let(:persister)`' unless defined? persister
+    class CustomResource < Valkyrie::Resource
+      include Valkyrie::Resource::AccessControls
+      attribute :id, Valkyrie::Types::ID.optional
+      attribute :authors, Valkyrie::Types::Array.optional.meta(ordered: true)
+    end
+  end
+  after do
+    Object.send(:remove_const, :CustomResource)
+  end
+  subject { persister }
+  let(:resource_class) { CustomResource }
+  let(:resource) { resource_class.new }
+  it "saves ordered properties and returns them in the appropriate order" do
+    resource.authors = ["a", "b", "a"]
+    output = persister.save(resource: resource)
+    expect(output.authors).to eq ["a", "b", "a"]
```

Comment:
This looks good to me. Does re-querying for the same id and loading a new instance of the resource provide any better checks that the order is preserved correctly?

Reply:
I suspect that it would definitely be an improvement for test for that case.  It seems to be really dependent upon the ORM integrating into the `resource_factory` Object: https://github.com/samvera-labs/valkyrie/blob/master/lib/valkyrie/persistence/postgres/query_service.rb#L34 and https://github.com/samvera-labs/valkyrie/blob/master/lib/valkyrie/persistence/postgres/persister.rb#L17, but that integration is still vital.

Reply:
@jamesvanmil I tested this shared spec against Solr and it works fine. I'm inclined to call this "good enough" and move on.